aten/src/TH/THAllocator.cpp

```diff
@@ -55,7 +55,7 @@ struct THMapAllocatorContext_ {
 #define TH_ALLOC_ALIGNMENT 64
 
 typedef struct {
-  int refcount;
+  std::atomic<int> refcount;
 } THMapInfo;
```

Comment:
Can we also add an assert that `sizeof(THMapInfo) < TH_ALLOC_ALIGNMENT`? I'm afraid if `std::atomic` ends up being not lock free on some platform, then it will use up more than 64B, because IIRC that's how big is `std::mutex`. That will effectively make the mutex or refcount alias with the data and it will be a disaster